testing/run_tests.py

```diff
@@ -57,63 +57,67 @@ def FindExecutablePath(path):
 def RunEngineExecutable(build_dir, executable_name, filter, flags=[], cwd=buildroot_dir):
   if filter is not None and executable_name not in filter:
     print 'Skipping %s due to filter.' % executable_name
-    return
+    return 0
 
   executable = FindExecutablePath(os.path.join(build_dir, executable_name))
 
   print 'Running %s in %s' % (executable_name, cwd)
   test_command = [ executable ] + flags
   print ' '.join(test_command)
-  subprocess.check_call(test_command, cwd=cwd)
+  return subprocess.check_call(test_command, cwd=cwd)
```

Comment:
Why isn't this raising an exception on non-zero exit codes?

Also, how convinced are we that processes won't return positive and/or negative values that end up cancelling each other out (e.g. one test returns 1 and another returns -1 to signal error)?

Reply:
Ah, I missed this. It looks like this is how the script has been working, but the JUnit tests were using the wrong thing and weren't asserting like the others. Changed the patch to fix the missing `check_call`.